example/passthrough.c

```diff
@@ -521,6 +521,24 @@ static off_t xmp_lseek(const char *path, off_t off, int whence, struct fuse_file
 	return res;
 }
 
+#ifdef linux
+static int xmp_statx(const char *path, int flags, int mask, struct statx *stxbuf,
+		     struct fuse_file_info *fi)
+{
+	int fd = 0;
+	int res;
+
+	if (fi)
+	    fd = fi->fh;
+
+	res = statx(fd, path, flags | AT_SYMLINK_NOFOLLOW, mask, stxbuf);
```

Comment:
Kernel code has
```
	if (file && S_ISREG(inode->i_mode)) {
		struct fuse_file *ff = file->private_data;

		inarg.getattr_flags |= FUSE_GETATTR_FH;
		inarg.fh = ff->fh;
	}
```

So shouldn't be an empty path used when fi is set, with the AT_EMPTY_PATH flag?

from `man 2 statx`
       By file descriptor
              If pathname is an empty string and the AT_EMPTY_PATH flag is specified in flags  (see  be‐
              low), then the target file is the one referred to by the file descriptor dirfd.

Please correct me if I'm wrong, but right now it would treat `fd` as dirfd?

Reply:
Also initialization `fd = 0` is a bit weird - 0 is a valid random value.

Reply:
Ok, fd is ignored when pathname begins with a slash, but could you still initialize it to -1?

Reply:
I believe lib/fuse.c (in line 4475 below) will automatically set AT_EMPTY_PATH in flags if the path is NULL, so that the fuse server (if it's not using the low-level interface) doesn't have to worry about it. In lib/fuse.c 's statx implementation, if fi is set it calls get_path_nullok() (in line 4467) which can either return a NULL path or return a valid path.